instrumentation/github.com/aws/aws-sdk-go-v2/otelaws/aws.go

```diff
@@ -0,0 +1,115 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelaws
+
+import (
+	"context"
+
+	v2Middleware "github.com/aws/aws-sdk-go-v2/aws/middleware"
+	"github.com/aws/smithy-go/middleware"
+	smithyhttp "github.com/aws/smithy-go/transport/http"
+
+	"go.opentelemetry.io/contrib"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/semconv"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	tracerName = "go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws"
+)
+
+type oTelMiddlewares struct {
+	tracer trace.Tracer
+}
+
+func (m oTelMiddlewares) spanCreator(stack *middleware.Stack) error {
+	return stack.Initialize.Add(middleware.InitializeMiddlewareFunc("OtelSpanCreator", func(
+		ctx context.Context, in middleware.InitializeInput, next middleware.InitializeHandler) (
+		out middleware.InitializeOutput, metadata middleware.Metadata, err error) {
+
+		opts := []trace.SpanOption{
+			trace.WithSpanKind(trace.SpanKindClient),
+		}
+		ctx, span := m.tracer.Start(ctx, "AWS", opts...)
```

Comment:
Good to know we need to consider `head-based samplers` case. 
But starting span in `After` will miss an edge case, which is if the error is happened in a middleware before.

Reply:
To do that the span would need to be ended in the `middleware.Before` middleware.  That would mean not ending the span in the `middleware.After` middleware, but putting it in the `metadata` so that it can be extracted and ended in the `middleware.Before` middleware.

Reply:
that involves complexity. From my test trace.SpanFromContext(ctx) cannot retrieve span generated in downstream.
And currently aws sdk go v2 has an [issue](https://github.com/aws/aws-sdk-go-v2/issues/1154) that cannot pass metadata up. 
Now It is acceptable that we only loss a part of shutdown time.